mars/tensor/expressions/indexing/setitem.py

```diff
@@ -30,41 +31,40 @@ class TensorIndexSetValue(IndexSetValue, TensorOperandMixin):
     def __init__(self, dtype=None, sparse=False, **kw):
         super(TensorIndexSetValue, self).__init__(_dtype=dtype, _sparse=sparse, **kw)
 
-    def _handle_inputs(self, inputs):
-        if len(inputs) == 1:
-            # get indexes and value from existed IndexOp
-            tensor = inputs[0]
-            indexes = self._indexes
-            value = self._value
-        else:
-            tensor, indexes, value = inputs
+    @contextlib.contextmanager
+    def _handle_params(self, inputs, indexes, value):
+        if indexes is not None and value is not None:
+            indexes_inputs = [ind for ind in indexes if isinstance(ind, TENSOR_TYPE + CHUNK_TYPE)]
+            inputs += indexes_inputs
+            if isinstance(value, TENSOR_TYPE + CHUNK_TYPE):
+                inputs += [value]
+        yield inputs
+
+        if indexes is not None:
             self._indexes = indexes
```

Comment:
Should these self._indexes and self._value setter move before `yield inputs`? Normally the `_set_inputs` which called in `new_tensors` and `new_chunks` ensure no Tensor/Chunk come in.

Better to add ut in tensor/expressions that no Tensor/Chunk come in.

Reply:
`_set_inputs` only handle the `_inputs` attribute, `_indexes` and `_value` will be set by iterating through `self._inputs`, so the behavior is acceptable i think.